ui/shared/components/form/ReduxFormWrapper.jsx

```diff
@@ -114,30 +115,49 @@ export const configuredField = (field, formProps = {}) => {
     fieldComponent: component,
     submitForm: formProps.submitOnChange ? formProps.onSubmit : null,
     label: helpLabel(label, labelHelp),
+    data: formProps.formMetaId && { formId: formProps.formMetaId },
     ...fieldProps,
   }
   return isArrayField ? (
-    <FieldArray
-      {...baseProps}
-      component={arrayFieldItem({ addArrayElement, addArrayElementProps, arrayFieldName, singleFieldProps, label })}
-    />
+    <FieldArray {...baseProps}>
+      {({ fields }) => (
+        <ArrayFieldItem
+          fields={fields}
+          addArrayElement={addArrayElement}
+          addArrayElementProps={addArrayElementProps}
+          arrayFieldName={arrayFieldName}
+          singleFieldProps={singleFieldProps}
+          label={label}
+        />
+      )}
+    </FieldArray>
   ) : <Field {...baseProps} {...singleFieldProps} />
 }
 
 export const configuredFields = props => props.fields.map(field => configuredField(field, props))
 
-class ReduxFormWrapper extends React.Component {
+const SUBMISSION_PANEL_SUBSCRIPTION = [
+  'submitSucceeded',
+  'submitFailed',
+  'dirty',
+  'dirtySinceLastSubmit',
+  'hasSubmitErrors',
+  'hasValidationErrors',
+  'errors',
+  'submitErrors',
+].reduce((acc, k) => ({ ...acc, [k]: true }), {})
+const SUBMITTING_SUBSCRIPTION = { submitting: true }
+const SUBMIT_SUCCEEDED_SUBSCRIPTION = { submitSucceeded: true }
+const DIRTY_SUBSCRIPTION = { dirty: true }
+
+class ReduxFormWrapper extends React.PureComponent {
```

Comment:
agreed. I did it in this separate PR because it made this PR virtually unreadable, but my thought was to release them both together: https://github.com/broadinstitute/seqr/pull/2503